OpenRA.Game/Graphics/Vertex.cs

```diff
@@ -15,24 +15,19 @@ namespace OpenRA.Graphics
 	[StructLayout(LayoutKind.Sequential)]
 	public struct Vertex
 	{
-		// TODO Workaround for unused field warnings in mono 2.10
-		#pragma warning disable 414
-		float x, y, z, u, v;
-		float p, c;
-		#pragma warning restore
+		public readonly float X, Y, Z, U, V, P, C;
```

Comment:
I mean we should be packing the vertices better. floats are overkill. Half-float, unsigned byte, 10/10/10/2 are all suitable options. Big vertices carry a hefty memory bandwidth cost.